### DISCUSSION TITLE
AMREX_ALWAYS_ASSERT that works on device?

### DISCUSSION THREAD

**Discussion by BenWibking** [AMReX-Codes/amrex#2648] [Q&A] [answered]
Is there a standard way to abort the code from a GPU kernel? I have a cell-by-cell iterative solve that should trigger a code abort if it fails to converge but I can't find a macro that will do this from the device side.

**Comment by WeiqunZhang** [comment, marked as answer]
Yes, `AMREX_ALWAYS_ASSERT` works on device if `NDEBUG` is not defined.  This is the same behavior as `assert` in CUDA. https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#assertion

**Reply by BenWibking** [reply]
Ah... CMake was adding -DNDEBUG to the compile flags when set to a Release build. Thanks.
